src/GridFieldNestedFormItemRequest.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Symbiote\GridFieldExtensions;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\Director;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldAddNewButton;
+use SilverStripe\Forms\GridField\GridFieldConfig_RecordEditor;
+use SilverStripe\Forms\GridField\GridFieldDataColumns;
+use SilverStripe\Forms\GridField\GridFieldDetailForm;
+use SilverStripe\Forms\GridField\GridFieldDetailForm_ItemRequest;
+use SilverStripe\Forms\GridField\GridFieldEditButton;
+use SilverStripe\Forms\GridField\GridFieldFilterHeader;
+use SilverStripe\Forms\GridField\GridFieldPageCount;
+use SilverStripe\Forms\GridField\GridFieldSortableHeader;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\HasManyList;
+use SilverStripe\ORM\Hierarchy\Hierarchy;
+use SilverStripe\View\ArrayData;
+use Symbiote\GridFieldExtensions\GridFieldAddNewInlineButton;
+use Symbiote\GridFieldExtensions\GridFieldEditableColumns;
+use Symbiote\GridFieldExtensions\GridFieldOrderableRows;
+
+class GridFieldNestedFormItemRequest extends GridFieldDetailForm_ItemRequest
+{
+    
+    public function Link($action = null)
+    {
+        return Controller::join_links($this->component->Link($this->record->ID), $action);
+    }
+    
+    public function ItemEditForm()
```

Comment:
One reason the gridfield config is created where it is now is that in the default config, we sometimes want to check permissions on the specific record to see what components should be added. For example the default config is based on `GridFieldConfig_RecordEditor`, but if the user can't edit a given record, then that record's nested gridfield will not include the add button. This won't be possible in the same way if we move the logic to `GridFieldNestedForm`, since then the config is the same for all records. For now we have found it more usable to us to configure the nested gridfield config directly in the DataObject classes, or via extensions to said classes.
I can refactor it this way, but then we need to deside if we do any of these permission checks at all, since they might then be confusing for the developer.

Reply:
Ahh I've just noticed the `getNestedConfig()` check there which solves some of this. That should definitely be added to the documentation (which still needs to be added to as per https://github.com/symbiote/silverstripe-gridfieldextensions/pull/384#pullrequestreview-2013562592 "Please add additional documentation so I know what needs to be manually tested and so that developers know what functionality this thing has.").

That's probably okay for now, given there is at least a per-model level of control.